File: javav2/example_code/route53/src/main/java/com/example/route/GetHealthCheckStatus.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.route;

// snippet-start:[route53.java2.get_health_check_status.main]
// snippet-start:[route53.java2.get_health_check_status.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.route53.Route53Client;
import software.amazon.awssdk.services.route53.model.GetHealthCheckStatusRequest;
import software.amazon.awssdk.services.route53.model.GetHealthCheckStatusResponse;
import software.amazon.awssdk.services.route53.model.HealthCheckObservation;
import software.amazon.awssdk.services.route53.model.Route53Exception;
import java.util.List;
// snippet-end:[route53.java2.get_health_check_status.import]

/**
 * Before running this Java V2 code example, set up your development
 * environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class GetHealthCheckStatus {
    public static void main(String[] args) {
        final String usage = """

                Usage:
                    <healthCheckId>\s

                Where:
                    healthCheckId - The health check id.\s
                """;

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String healthCheckId = args[0];
        Region region = Region.AWS_GLOBAL;
        Route53Client route53Client = Route53Client.builder()
                .region(region)
                .build();

        getHealthStatus(route53Client, healthCheckId);
        route53Client.close();
    }

    public static void getHealthStatus(Route53Client route53Client, String healthCheckId) {
        try {
            GetHealthCheckStatusRequest statusRequest = GetHealthCheckStatusRequest.builder()
                    .healthCheckId(healthCheckId)
                    .build();

            GetHealthCheckStatusResponse statusResponse = route53Client.getHealthCheckStatus(statusRequest);
            List<HealthCheckObservation> observations = statusResponse.healthCheckObservations();
            for (HealthCheckObservation observation : observations) {
                System.out.println("(The health check observation status: " + observation.statusReport().status());
            }

        } catch (Route53Exception e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}
// snippet-end:[route53.java2.get_health_check_status.main]

File: .kotlin_alpha/services/route53/src/main/kotlin/com/kotlin/route/GetHealthCheckStatus.kt
// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
// snippet-sourcedescription:[GetHealthCheckStatus.kt demonstrates how to get the status of a specific health check.]
// snippet-keyword:[AWS SDK for Kotlin]
// snippet-service:[Amazon Route 53]
// snippet-keyword:[Code Sample]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[11/5/2021]
// snippet-sourceauthor:[AWS - scmacdon]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.route

// snippet-start:[route53.kotlin.get_health_check_status.import]
import aws.sdk.kotlin.services.route53.Route53Client
import aws.sdk.kotlin.services.route53.model.GetHealthCheckStatusRequest
import aws.sdk.kotlin.services.route53.model.Route53Exception
import kotlin.system.exitProcess
// snippet-end:[route53.kotlin.get_health_check_status.import]

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <healthCheckId> 

    Where:
         healthCheckId - the health check id.
    """

      if (args.size != 1) {
          println(usage)
          exitProcess(0)
      }

    val healthCheckId = args[0]
    val route53Client = Route53Client{region = "AWS_GLOBAL"}
    getHealthStatus(route53Client, healthCheckId)
    route53Client.close()
}

// snippet-start:[route53.kotlin.get_health_check_status.main]
suspend  fun getHealthStatus(route53Client: Route53Client, healthCheckIdVal: String?) {
        try {
            val statusRequest = GetHealthCheckStatusRequest {
                healthCheckId = healthCheckIdVal
            }

            val response = route53Client.getHealthCheckStatus(statusRequest)
            response.healthCheckObservations?.forEach { observation ->
                   println("(The health check observation status is ${observation.statusReport?.status}")
            }

        } catch (e: Route53Exception) {
            System.err.println(e.message)
            exitProcess(0)
        }
 }
// snippet-end:[route53.kotlin.get_health_check_status.main]